### DISCUSSION TITLE
Inflated doc_bin size when piping as_tuples

### DISCUSSION THREAD

**Discussion by langdonholmes** [explosion/spaCy#7486] [Help: Coding & Implementations] [answered]
I am using the en_core_web_trf-3.0.0 model without a GPU, if that matters. 

```
def proc_texts(texttuples):
    doc_bin = DocBin(attrs=["ORTH", "TAG", "HEAD", "DEP", "LEMMA", "MORPH", "POS"], store_user_data=True)
    for doc, name in nlp.pipe(texttuples, as_tuples=True):
        doc._.name = name
        doc_bin.add(doc)
    return doc_bin
```

Results in a filesize of 5GB. The raw text corpus is quite small at 7MB. Here is my method for compiling the list of (text, name) tuples:

```
def getcorpus(corpusdir):
    texts = []
    for file in Path(corpusdir).rglob("**\*.txt"):
        with open(file, 'r', encoding="utf-8-sig") as f:
            data = f.read()
            data = re.sub(r"\s+", r" ", data) 
            texts.append((data, file.name))
    return texts
```

I suspect this is an issue, but I'm not sure how to properly report an issue, so I thought I would post here first. The resulting 5GB doc_bin appears to function totally normally. I am wondering if the 'vocab' is being attached to every doc in the bin separately? There are about 6,000 texts, each about 200 words.

Happy to provide any additional information.

![image](https://user-images.githubusercontent.com/55119338/111676071-05cf6e80-87db-11eb-9a38-c2a90902b847.png)

**Comment by adrianeboyd** [comment, marked as answer]
Hi, I don't think `as_tuples` or your `._.name` extension is the culprit. The `trf` models save transformer tokens+output as `doc._.trf_data`, which is included as part of the user data with `store_user_data=True` and can be much larger than the rest of the doc annotation. You can manually remove it from the docs before adding them to the DocBin with:

```python
doc._.trf_data = None
```

**Reply by langdonholmes** [reply]
Adriane, thanks so much for your explanation and solution!

It would be nice if we could turn this feature off when loading the model:

```Python
spacy.load(some-model-trf, set_annotations = Bool, set_extra_annotations = Callable[List])
```

Edit: doc_bin filesize is down to 6MB using `doc._.trf_data = None` -- wonderful.

**Reply by adrianeboyd** [reply]
It's not easy to disable this automatically in the `transfomer` config because the later pipeline components that "listen" to the transformer need this information to be stored in the doc.

However, at the end of the pipeline (when you know that all components that might use this data have run), you can add a custom component that removes it. It could look something like this:

```python
from spacy.language import Language

@Language.component("remove_trf_data")
def remove_trf_data(doc):
    doc._.trf_data = None
    return doc


nlp.add_pipe("remove_trf_data")
```

**Reply by langdonholmes** [reply]
This seems like an even better solution. Thanks again.

**Comment by kaushikepi** [comment]
@langdonholmes Does add the above custom component will release GPU memory after using doc object output.
I have tried adding it and monitored the GPU uses by inferencing text files on the go to extract NER entity using the 'en_core_web_trf' model.
Instead adding `torch.cuda.empty_cache()` helped
